pkg/sql/logictest/tests/local-mixed-25.4/local-mixed-25_4_test: TestLogic_row_level_security failed

pkg/sql/logictest/tests/local-mixed-25.4/local-mixed-25_4_test.TestLogic_row_level_security [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20799445?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20799445?buildTab=artifacts#/) on master @ [70ac59b132e1249c20c534553ec4f6d8b71487dc](https://github.com/cockroachdb/cockroach/commits/70ac59b132e1249c20c534553ec4f6d8b71487dc):


```
=== RUN   TestLogic_row_level_security
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/bffe045638cb29b973425866ff181950/logTestLogic_row_level_security2691299572
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:121: cluster virtualization disabled in global scope due to issue: #156124 (expected label: C-bug)
[20:23:00] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16643/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-25.4/local-mixed-25_4_test_/local-mixed-25_4_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/row_level_security with config local-mixed-25.4: 906 tests, 2 failures
[20:23:01] --- total progress: 906 statements
--- total: 906 tests, 2 failures
    logic.go:4728: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/bffe045638cb29b973425866ff181950/logTestLogic_row_level_security2691299572
--- FAIL: TestLogic_row_level_security (74.61s)
=== RUN   TestLogic_row_level_security/bypassrls
    logic.go:3201: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16643/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-25.4/local-mixed-25_4_test_/local-mixed-25_4_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/row_level_security:3971: SELECT * FROM bypassrls ORDER BY id;
        expected:
            0 visible: 0
            2 visible: 2
            
        but found (query options: "") :
            0   visible: 0
            1   hidden: 1
            2   visible: 2
            10  hidden: value
            11  hidden: value
            20  hidden: value
            30  hidden: value
            31  hidden: value
            40  hidden: value
            
    logic.go:2382: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16643/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-25.4/local-mixed-25_4_test_/local-mixed-25_4_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/row_level_security:3977: too many errors encountered, skipping the rest of the input
--- FAIL: TestLogic_row_level_security/bypassrls (1.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_row_level_security.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57419

--- a/pkg/sql/catalog/lease/lease.go
+++ b/pkg/sql/catalog/lease/lease.go
@@ -1282,6 +1282,11 @@ func (m *Manager) acquireNodeLease(
 				if err := m.upsertDescriptorIntoState(ctx, id, session, desc); err != nil {
 					return false, err
 				}
+				// Descriptor versions for these tables participate directly in memo
+				// staleness checks. Bump the lease generation now so queries that
+				// rely on the new privileges/options are forced to replan immediately
+				// instead of waiting for the asynchronous lease update.
+				m.leaseGeneration.Add(1)
 
 				desc, err = doAcquisition()
 				if err != nil {

